File: parser/SyntaxErrorListener.js
const antlr4 = require('antlr4'),
      ErrorListener = antlr4.error.ErrorListener;

class SyntaxErrorListener extends ErrorListener {
    constructor() {
        super();
        this.messages = [];
    }

    syntaxError(recognizer, offendingSymbol, line, column, message, e) {
        this.messages.push({
            message,
            line,
            column
        });
    }
}

module.exports = SyntaxErrorListener;

File: src/cpp/SyntaxErrorListener.cpp
#include "SyntaxErrorListener.h"

using namespace antlr4;
using namespace std;

SyntaxErrorListener::SyntaxErrorListener(vector<SyntaxError> *errorList) : errors(errorList) {}

void SyntaxErrorListener::syntaxError(Recognizer *recognizer,
                                      Token *offendingSymbol,
                                      size_t line,
                                      size_t charPositionInLine,
                                      const string &msg,
                                      exception_ptr e)
{
    errors->push_back(SyntaxError{msg, line, charPositionInLine});
}